build-in-container/README.md

```diff
@@ -0,0 +1,45 @@
+# Build-in-container
+The build-in-container tool provides a developer tool to quickly build Mariner packages. It is easy-to-use, and distribution and platform agnostic. It sets up a build environment in an expedient manner using a container.
+
+Please install docker on your system before using the tool.
+
+## Usage
+The run.sh script presents these options <br />
+-t                 creates container image <br />
+-b [repo_dir]      creates container, builds Mariner and outputs to out/ <br />
+-i [repo_dir]      create an interactive Mariner build container <br />
+-c [repo_dir]      cleans up the current workspace <br />
+--help             shows help on usage <br /> <br />
+
+- Unless provided, repo_dir defaults to the directory containing build-in-container tool <br />
+- Place specs to build under $repo_dir/SPECS/ <br />
+- The output from the build will be available under $repo_dir/out/ (RPMS and SRPMS) <br />
+- Logs are published under $repo_dir/logs/ <br />
+
+## Details on what goes on inside the container:
+### Creating container image
+'create-build-container.sh' creates an image that the docker can use to launch the Mariner build container. It downloads a Mariner2.0 container image, and makes suitable modifications to it. The output image is tagged as 'mcr.microsoft.com/mariner-container-build:2.0'
+
+### Running container in the specified mode
+'mariner-docker-run.sh' starts a docker container using the image produced in Step(1). 
+
+In the _build_ mode, it sets up the Mariner build system inside the container, builds all the specs under $repo_dir/SPECS/ and outputs to $repo_dir/out/.
+
+In the _interactive_ mode, it sets up the Mariner build system inside the container, and starts the container at /mariner/toolkit/. The user can invoke Mariner `make` commands to build packages, images and more. Please see the [section](https://github.com/microsoft/CBL-MarinerTutorials/tree/main/buildInContainer/build-in-container#sample-make-commands) for sample `make` commands, and visit [Mariner Docs](https://github.com/microsoft/CBL-Mariner/blob/2.0/toolkit/docs/building/building.md) for the complete set of commands. 
```

Comment:
I'd call out that it still uses the stable toolkit rather than a toolkit that might exist inside the folder you are running this from.

Reply:
If given a repo_dir, it'll take the toolkit (and all Mariner folders) from there. Else, it'll take toolkit from the build-in-container repo. It is not recognising the folder one is running from, because that need not be a valid mariner repo.

Reply:
Ah, I missed that. I'd still add something to that affect into the readme though.